src/components/app-chips-list/AppChipList.js

```diff
@@ -1,34 +1,45 @@
-import { useState } from 'react'
-import { Chip, Box } from '@mui/material'
+import Box from '@mui/material/Box'
+import Chip from '@mui/material/Chip'
+
+import AppChip from '~/components/app-chip/AppChip'
+import AppPopover from '~/components/app-popover/AppPopover'
+
 import { styles } from '~/components/app-chips-list/AppChipsList-styles'
-import AppChip from '../app-chip/AppChip'
 
-const AppChipList = ({ items, defaultQuantity, handleChipDelete }) => {
-  const [isOpen, setIsOpen] = useState(false)
-  const amountOfChips = isOpen ? items.length : defaultQuantity
-  const itemsForChips = items.slice(0, amountOfChips)
+const AppChipList = ({ items, defaultQuantity, handleChipDelete = null, icon }) => {
   const hideChips = items.length - defaultQuantity > 0 && items.length - defaultQuantity
 
-  const chips = itemsForChips.map((item) => (
-    <AppChip handleDelete={ () => handleChipDelete(item) } key={ item }>
-      { item }
-    </AppChip>
-  ))
+  const chips = items.map((item) =>
+    handleChipDelete ? (
+      <AppChip handleDelete={ () => handleChipDelete(item) } key={ item }>
```

Comment:
move this to props const, const props = handleChipDelete ? { handleDelete :  () => handleChipDelete(item) } : { icon }
<AppChip {...props} key={ item }>